### PR TITLE
[GPU] Sliding window support for non-PA SDPA

### PR DESCRIPTION
**Pull request by vmadananth**
### Details:
Adds sliding-window (SWA) attention support to the non-paged-attention SDPA path on GPU,
wires it through from `GroupQueryAttention` decomposition.

## Changes

**Sliding window plumbing (op → kernel)**
- Add `sliding_window` attribute to `op::SDPA` and the `scaled_dot_product_attention`
  cldnn primitive, propagated through `sdpa_configuration` into JIT emission
- `GroupQueryAttentionDecomposition::make_sdpa` now sets `sliding_window_size` on the
  generated `op::SDPA` node when a local window is configured.
- `make_attention_mask` now also skips building the explicit additive-mask subgraph when
  `sliding_window_cache && causal` (in addition to the existing "no window at all" skip
  case), since the kernel applies the window natively via `is_causal` +
  `sliding_window_size` — avoids redundant mask computation whenever that path is reachable
  

### Tickets:
 - *ticket-id*

### AI Assistance:
 - *AI assistance used:  yes*
 - Test cases written by AI 
 
1. sdpa_sliding_window_test.basic (parameterized, {head_size, num_heads, seq_q, seq_kv, batch, window}) — compares sdpa_micro (causal + sliding_window attribute) against sdpa_ref with an explicit mask as ground truth, asserting cosine similarity ≥ 0.95. 
Cases:
- {64, 8, 128, 128, 1, 32} / {64, 8, 128, 128, 1, 128} — equal-length seq, window smaller/equal to sequence.
- {64, 8, 64, 64, 1, 256} — window larger than the sequence (no effective clipping).
- {64, 8, 1, 512, 1, 128} — single-token decode against a long KV history.
- {64, 8, 512, 512, 1, 64} — larger equal-length sequence.
- {128, 4, 256, 256, 2, 64} — larger head size, batch=2.
- {64, 8, 128, 768, 1, 512} — the chunked-prefill regression case: 128 new tokens on top of 640 prior KV tokens, window smaller than total history

2. sdpa_sliding_window_test.causal_true_with_explicit_mask — checks that passing is_causal=true together with an explicit attention_mask input produces the same result as is_causal=false + explicit mask (cosine similarity ≥ 0.95), verifying sdpa_micro's WITH_ATTN_MASK path behaves consistently when both are set.

3. sdpa_sliding_window_op_test.attribute_preserved_on_clone / default_is_zero — pure op-level (no GPU execution): verify sliding_window_size survives clone_with_new_inputs/visit_attributes round-trip, and defaults to 0 when unset.

### CODE REVIEW

File: src/plugins/intel_gpu/src/plugin/transformations/expand_broadcast_reshape_sdpa_fusion.cpp [hunk 1]
```diff
@@ -385,6 +385,7 @@ ExpandBroadcastReshapeSDPAFusion::ExpandBroadcastReshapeSDPAFusion() {
                                order_d,
                                ov::element::dynamic,
                                sdpa->get_causal_mask_alignment());
+        sdpa_new->set_sliding_window_size(sdpa->get_sliding_window_size());
```
Comment (Kotomi-Du) [inline src/plugins/intel_gpu/src/plugin/transformations/expand_broadcast_reshape_sdpa_fusion.cpp:388]: The current setter pattern makes it easy to lose the attribute when another transformation reconstructs the node. 
please consider use `clone_with_new_inputs()` to reserve the attribute

File: src/plugins/intel_gpu/include/intel_gpu/op/sdpa.hpp [hunk 2]
```diff
@@ -60,6 +60,9 @@ class SDPA : public ov::op::v13::ScaledDotProductAttention {
     QuantizationAttribute get_quantization_attrs() const { return m_quantization_attrs; }
     size_t get_compression_inputs_num() const;
 
+    int64_t get_sliding_window_size() const { return m_sliding_window_size; }
+    void set_sliding_window_size(int64_t v) { m_sliding_window_size = v; }
```
Comment (Kotomi-Du) [inline src/plugins/intel_gpu/include/intel_gpu/op/sdpa.hpp:64]: please add validation for the new attribute: 
1. sliding_window_size >= 0;
2. decide whether nonzero requires is_causal;
3. document that 0 means disabled;

File: src/plugins/intel_gpu/src/plugin/transformations/group_query_attention_decomposition.hpp [hunk 3]
```diff
@@ -33,6 +33,8 @@ class GroupQueryAttentionDecomposition : public ov::pass::GroupQueryAttentionDec
                                                   bool sliding_window_cache,
                                                   float scale,
                                                   bool has_sink) override;
+
+    int64_t m_local_window_size = -1;
```
Comment (Kotomi-Du) [inline src/plugins/intel_gpu/src/plugin/transformations/group_query_attention_decomposition.hpp:37]: consider pass the parameter directly from maks_sdpa parameter, rather than add a new parameter in this file. 

```
 std::shared_ptr<ov::Node> GroupQueryAttentionDecomposition::make_sdpa(const ov::Output<ov::Node>& query,
                                                                       const ov::Output<ov::Node>& key,
                                                                       const ov::Output<ov::Node>& value,
                                                                       const ov::Output<ov::Node>& mask,
                                                                       const ov::Output<ov::Node>& scale,
                                                                       const ov::Output<ov::Node>& sink,
-                                                                      bool is_causal) {
+                                                                      bool is_causal,
+                                                                      int64_t local_window_size) {
@@
-    const auto order = op::SDPA::default_order(query.get_partial_shape().rank().get_length());
-    return register_new_node<op::SDPA>(inputs,
+    const auto order = op::SDPA::default_order(query.get_partial_shape().rank().get_length());
+    auto sdpa = register_new_node<op::SDPA>(inputs,
                                        is_causal,
                                        order,
                                        order,
                                        order,
                                        order,
                                        ov::element::dynamic,
                                        is_causal ? op::SDPA::CausalMaskAlignment::LOWER_RIGHT : op::SDPA::CausalMaskAlignment::UPPER_LEFT);
+    if (local_window_size >= 1) {
+        sdpa->set_sliding_window_size(local_window_size);
+    }
+    return sdpa;```

File: src/plugins/intel_gpu/src/graph/impls/ocl_v2/sdpa/sdpa_base.hpp [hunk 4]
```diff
@@ -52,6 +52,8 @@ struct sdpa_configuration {
     bool has_score_aggregation = false;
     bool has_rotated_blocks = false;
 
+    int64_t sliding_window = 0;
```
Comment (Kotomi-Du) [inline src/plugins/intel_gpu/src/graph/impls/ocl_v2/sdpa/sdpa_base.hpp:55]: GQA operator set sliding_windows =-1 as  disable, shall we make it consistent as well?

**Review by Kotomi-Du — COMMENTED** [review]
please check my comments
